core/src/main/java/smashdudes/ecs/systems/CharacterJumpInputSystem.java

```diff
@@ -30,13 +26,32 @@ protected void updateEntity(Entity entity, float dt)
         CharacterInputComponent i = entity.getComponent(CharacterInputComponent.class);
         VelocityComponent v = entity.getComponent(VelocityComponent.class);
         JumpComponent j = entity.getComponent(JumpComponent.class);
+        InAirComponent a = entity.getComponent(InAirComponent.class);
+        OnGroundComponent g = entity.getComponent(OnGroundComponent.class);
 
-        if(i.currentState.up)
+        if(i.currentState.up && !i.previousState.up)
         {
-            v.velocity.y = j.jumpStrength;
-            j.disable();
+            if (j.remainingJumps == j.maxJumps && g.isEnabled())
+            {
+                v.velocity.y = j.jumpStrength;
+                a.enable();
+                g.disable();
+            }
+            else if (j.remainingJumps == j.maxJumps)
```

Comment:
This should be simplified. Notice how the first if and the else if share conditionals. Should be re-written as
`
if(j.remainingJumps == j.maxJumps)
{
  if(g.isEnabled())
  {
  }
  else
  {
  }
}
`

Reply:
good catch